File: backend/app/models/politician.py
import uuid
from datetime import datetime

from sqlalchemy import DateTime, String, Text, func
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.database import Base


class Politician(Base):
    __tablename__ = "politicians"

    id: Mapped[uuid.UUID] = mapped_column(primary_key=True, default=uuid.uuid4)
    slug: Mapped[str] = mapped_column(String(120), unique=True, index=True)
    full_name: Mapped[str] = mapped_column(String(255))
    biography: Mapped[str | None] = mapped_column(Text)
    image_url: Mapped[str | None] = mapped_column(String(500))
    is_deleted: Mapped[bool] = mapped_column(default=False)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())

    memberships = relationship("PartyMembership", back_populates="politician")
    statements = relationship("Statement", back_populates="politician")

File: frontend/src/types/politician.ts
import type { PoliticalParty } from "./politicalParty";
import type { CriteriaAverages } from "./aiAnalysis";
import type { StatementListItem } from "./statement";

export type Politician = {
  id: string;
  slug: string;
  full_name: string;
  biography?: string | null;
  image_url?: string | null;
  current_party?: PoliticalParty | null;
  average_scores?: CriteriaAverages | null;
  statements?: StatementListItem[];
};
